packages/flutter_tools/lib/src/base/os.dart

```diff
@@ -44,6 +44,11 @@ abstract class OperatingSystemUtils {
   File makePipe(String path);
 
   void unzip(File file, Directory targetDirectory);
+
+  /// Returns the name of the [binaryName] executable.
+  /// 
+  /// No-op on most OS. Adds [winExtension] to the name on Windows.
```

Comment:
I was thinking about that. However, it doesn't works well here because I'd also have to set "exe" as the default value here [1] to make the pass-through work. I really want the following semantics: If you provide a valid non-null winExtension, that one will be used. Otherwise it is "exe". I Updated the doc comment to reflect that. Wdyt?

[1] https://github.com/flutter/flutter/pull/7622/files#diff-a47c8fbf82b59a5b53d91838b49d0232R248